File: m3project/src/main/java/sg/edu/ntu/m3project/m3project/repository/UserRepository.java
package sg.edu.ntu.m3project.m3project.repository;

import org.springframework.stereotype.Repository;

import java.util.Optional;

import org.springframework.data.repository.CrudRepository;

import sg.edu.ntu.m3project.m3project.entity.UserEntity;

@Repository
public interface UserRepository extends CrudRepository<UserEntity, Integer> {

    Optional<UserEntity> findByEmail(String email);

}

File: m3project/src/main/java/sg/edu/ntu/m3project/m3project/repository/UserRepository.kt
package sg.edu.ntu.m3project.m3project.repository

import org.springframework.data.repository.CrudRepository
import org.springframework.stereotype.Repository
import sg.edu.ntu.m3project.m3project.entity.UserEntity
import java.util.*

@Repository
interface UserRepository : CrudRepository<UserEntity?, Int?> {
    fun findByEmail(email: String?): Optional<UserEntity?>?
}